golos.publication/parameters.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <eosiolib/singleton.hpp>
+#include <common/parameter.hpp>
+#include <common/parameter_ops.hpp>
+
+namespace golos {
+
+    using namespace eosio;
+
+    struct st_max_vote_changes : parameter {
+        uint8_t max_vote_changes;
+
+        void validate() const override {
+            eosio_assert(max_vote_changes > 0, "Max vote changes can't be negative.");
+        }
+    };
+    using max_vote_changes_prm = param_wrapper<st_max_vote_changes, 1>;
+
+    struct st_cashout_window : parameter {
+        uint32_t cashout_window;
+
+        void validate() const override {
+            eosio_assert(cashout_window > 0, "Cashout window can't be negative.");
+        }
+    };
+    using cashout_window_prm = param_wrapper<st_cashout_window, 1>;
+
+    struct st_upvote_lockout : parameter {
+        uint8_t upvote_lockout;
+        
+        void validate() const override {
+            eosio_assert(upvote_lockout > 0, "Upvote lockout can't be negative.");
+        }
+    };
+    using upvote_lockout_prm = param_wrapper<st_upvote_lockout, 1>;
+
+    struct st_max_beneficiaries : parameter {
+        uint8_t max_beneficiaries;
+        
+        void validate() const override {
```

Comment:
There is no reason to disallow witnesses to disable beneficiaries at all, remove `validate()`